grakn-engine/src/main/java/ai/grakn/engine/tasks/manager/ZookeeperConnection.java

```diff
@@ -45,11 +53,15 @@
     private static final int ZOOKEEPER_CONNECTION_TIMEOUT = ConfigProperties.getInstance().getPropertyAsInt(ZK_CONNECTION_TIMEOUT);
     private final CuratorFramework zookeeperConnection;
 
+    private static final AtomicInteger CONNECTION_COUNTER = new AtomicInteger(0);
+
     /**
      * Start the connection to zookeeper. This method is blocking.
      */
-    public ZookeeperConnection(CuratorFramework zookeeperConnection) {
-        this.zookeeperConnection = zookeeperConnection;
+    public ZookeeperConnection() {
+        zookeeperConnection = client();
+
+        CONNECTION_COUNTER.incrementAndGet();
```

Comment:
It seems like here we are always starting the Zookeeper connection, but in the `close` method below, it only gets closed once. 
Do we need this counter? Why dont we try and use a singleton instead?